python/lsst/daf/butler/registry/interfaces/_versioning.py

```diff
@@ -180,19 +180,62 @@ def newSchemaVersion(self) -> VersionTuple | None:
 
         Notes
         -----
-        Default implementation only forks for extensions that support single
-        schema version and it returns version obtained from `currentVersions`.
-        If `currentVersions` returns multiple version then default
-        implementation will raise an exception and the method has to be
-        reimplemented by a subclass.
+        Extension classes which support multiple schema versions need to
+        override `_newDefaultSchemaVersion` method.
         """
-        my_versions = self.currentVersions()
+        return self.clsNewSchemaVersion(self._registry_schema_version)
+
+    @classmethod
+    def clsNewSchemaVersion(cls, schema_version: VersionTuple | None) -> VersionTuple | None:
+        """Class method which returns schema version for new registry.
+
+        Parameters
+        ----------
+        schema_version : `VersionTuple` or `None`
+            Configured schema version or `None` if default schema version
+            should be created. If not `None` then it is guaranteed to be
+            compatible with `currentVersions`.
+
+        Returns
+        -------
+        version : `VersionTuple` or `None`
+            Schema version created by this extension. `None` is returned if an
+            extension does not require its version to be saved or checked.
+
+        Notes
+        -----
+        Default implementation of this method can work in simple cases. If
+        the extension only supports single schema version than that versions is
+        returned. If the extension supports multiple schema versions and
+        ``schema_version`` is not `None` then ``schema_version`` is returned.
+        If the extension supports multiple schema versions, but
+        ``schema_version`` is `None` it calls ``_newDefaultSchemaVersion``
+        method which needs to be reimplemented in a subsclass.
+        """
+        my_versions = cls.currentVersions()
         if not my_versions:
             return None
         elif len(my_versions) == 1:
             return my_versions[0]
+        else:
+            if schema_version is not None:
+                assert schema_version in my_versions, "Schema version must be compatible."
```

Comment:
Do we get to this assertion if somebody tries to load a Butler whose software version is incompatible with the repo version?  If so, it probably shouldn't be an assert.

Reply:
This method is only used when creating new database, the non-None version that is passed as argument can come from `butler.yaml`, and there is a separate piece of code that verifies that the version is already compatible (and raises a better exception). I just added `assert` here as an additional protection in case someone decides to do something different.